hack/make/test-install-script

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# This script is used for testing install.sh and that it works for
+# each of component of our apt and yum repos
+set -e
+
+test_install_script(){
+	# these are equivalent to main, testing, experimental components
+	# in the repos, but its the url that will do the conversion
+	components=( experimental test get )
+
+	for component in "${components[@]}"; do
+		# change url to specific component for testing
+		local test_url=https://${component}.docker.com
+		local script=$(mktemp /tmp/install-XXXXXXXXXX.sh)
+		sed "s,url='https://get.docker.com/',url='${test_url}/'," hack/install.sh > "${script}"
+
+		chmod +x "${script}"
+
+		# test for each Dockerfile in contrib/builder
+		for dir in contrib/builder/*/*/; do
+			local from="$(awk 'toupper($1) == "FROM" { print $2; exit }' "$dir/Dockerfile")"
+
+			# change to  wheezy backports for init-system-helpers
+			if [ "$from" = "debian:wheezy" ]; then
+				from="debian:wheezy-backports"
```

Comment:
Ohh, we add backports in our `generate.sh`.  IMO it's worth changing that to instead just swap the base image at that level, and then this file just consumes that as-is.  Thoughts?